Document ImageVolume with an image diges

iholder101 opened the pull request:
### Description

Adds a small section under https://kubernetes.io/docs/tasks/configure-pod-container/image-volumes/ to document the new [KEP](https://github.com/kubernetes/enhancements/issues/5365) regarding reporting the image volume's digest as part of the pod's status.

### Description

Changes preview:
<img width="1075" height="932" alt="Screenshot From 2025-10-28 10-41-18" src="https://github.com/user-attachments/assets/2f293aab-3655-4fd6-87c2-c57cafab3a09" />


### Issue

KEP: https://github.com/kubernetes/enhancements/issues/5365.

iholder101 commented:
/uncc @lmktfy @divya-mohan0209

iholder101 commented:
> Hi @iholder101 thank you for opening this placeholder PR! Feature Blog PRs should be opened against the `main` branch as describe in our handbook [here](https://github.com/kubernetes/sig-release/tree/master/release-team/role-handbooks/communications?rgh-link-date=2025-10-23T07%3A52%3A46Z#feature-blog-pr-instructions-for-authors). Could you please fix the target branch in this PR?

Done! Thanks for raising my attention.

graz-dev commented:
Hi @iholder101 ,
I understood this PR was a placeholder for the feature blog ([ref](https://github.com/kubernetes/website/pull/52782#issuecomment-3430633228)) related to KEP https://github.com/kubernetes/enhancements/issues/5365.

I see you've now added docs content instead. Have you reconsidered writing the feature blog for KEP https://github.com/kubernetes/enhancements/issues/5365, or will you be opening a new PR for it?

Many thanks!

kannon92 reviewed (commented):
Can you remove dummy.txt?

iholder101 commented on content/en/docs/tasks/configure-pod-container/image-volumes.md:
```diff
@@ -100,6 +100,49 @@ from Kubernetes v1.33 when using the image volume feature.
 
+## Image volume with an image digest
+
+{{< feature-state feature_gate_name="ImageVolumeWithDigest" >}}
```
> It may be worth hyperlinking to those docs.

But this **is** the image volume docs, isn't it?

This is just a subsection in the image volume documentation, isn't it clear from the context that image volume needs to be enabled for the digest part to be active?

graz-dev commented:
> We've decided to include a small docs change instead of a blog post (xref https://github.com/kubernetes/enhancements/issues/5365#issuecomment-3435374431).

@iholder101 thanks, we are going to consider this KEP as opted-out for feature blogs.

dipesh-rawat commented on content/en/docs/reference/command-line-tools-reference/feature-gates/ImageVolumeWithDigest.md:
```diff
@@ -0,0 +1,14 @@
+    defaultValue: false
+    fromVersion: "1.35"
+---
+Report [ImageVolume](https://kubernetes.io/docs/tasks/configure-pod-container/image-volumes/)'s
```
We prefer to use relative links in doc where feasible.

```suggestion
Report [ImageVolume](/docs/tasks/configure-pod-container/image-volumes/)'s
```

dipesh-rawat reviewed (commented):
@iholder101  We should consider setting the `dev-1.35` branch as the target branch rather than the `main` branch for this pull request. Since the doc update is related to an Alpha feature in release v1.35

iholder101 committed:
[ImageVolume]: Add a small documentation regarding an image digest

Add a small note about KEP #5365 (about image volumes with an
image digest in the pod's status) in the Image Volume documentation:
https://kubernetes.io/docs/concepts/storage/volumes/#image

Signed-off-by: Itamar Holder <iholder@redhat.com>

iholder101 commented:
> /hold
> 
> This is dependent on [kubernetes/kubernetes#132807](https://github.com/kubernetes/kubernetes/pull/132807) merging.
> 
> Docs freeze is on December 2nd.

Implementation PR is now merged: https://github.com/kubernetes/kubernetes/pull/132807.
/unhold

In addition, this PR is now targeted at `main`.
Is there anything missing here?

lmktfy commented on content/en/docs/concepts/storage/volumes.md:
```diff
@@ -642,6 +642,49 @@ The following fields are available for the `image` type:
 See the [_Use an Image Volume With a Pod_](/docs/tasks/configure-pod-container/image-volumes)
 example for more details on how to use the volume source.
 
+#### Image volume with an image digest
```
```suggestion
#### Pod status and `image` volumes {#image-volume-pod-status}
```

lmktfy commented on content/en/docs/concepts/storage/volumes.md:
```diff
@@ -642,6 +642,49 @@ The following fields are available for the `image` type:
+After enabling the `ImageVolumeWithDigest` feature gate and deploying the pod
+as [explained above](#create-pod), you should see the following pod manifest.
+Note how the new `ImageRef` field under `volumeMounts` in the container status
+is now populated:
```
```suggestion
If the `ImageVolumeWithDigest` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/) is enabled in your cluster,
then whenever you specify an `image` volume for a Pod,
the kubelet updates the Pod status to record the _digest_
of the container image that's being used as a volume source.

lmktfy replied:
```suggestion
If the `ImageVolumeWithDigest` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/) is enabled in your cluster,
then whenever you specify an `image` volume for a Pod,
the kubelet updates the Pod status to record the _digest_
of the container image that's being used as a volume source.
```

This change aligns with our style advice to write [timeless documentation](https://developers.google.com/style/timeless-documentation).

lmktfy commented on content/en/docs/concepts/storage/volumes.md:
```diff
@@ -642,6 +642,49 @@ The following fields are available for the `image` type:
+as [explained above](#create-pod), you should see the following pod manifest.
+Note how the new `ImageRef` field under `volumeMounts` in the container status
+is now populated:
+
```
```suggestion

Here's a simplified example of a running Pod, represented as YAML, including the status update.

```
This is not really a manifest. You don't put `.status` in an object manifest; a manifest for a typical Kubernetes object has `.apiVersion`, `.kind`, `.metadata` and `.spec` only.

lmktfy commented on content/en/docs/concepts/storage/volumes.md:
````diff
@@ -642,6 +642,49 @@ The following fields are available for the `image` type:
+
+```yaml
+apiVersion: v1
+kind: Pod
````
Add `.metadata` as well, or include a comment to say that you have skilled it.

lmktfy commented on content/en/docs/concepts/storage/volumes.md:
```diff
@@ -642,6 +642,49 @@ The following fields are available for the `image` type:
+  containers:
+  - name: shell
+    command: ["sleep", "infinity"]
+    image: debian
```
(nit)
```suggestion
    image: docker.io/library/debian:12
```
We don't like to teach bad practice, such as running a `:latest` image.

lmktfy commented on content/en/docs/concepts/storage/volumes.md:
```diff
@@ -642,6 +642,49 @@ The following fields are available for the `image` type:
+  containerStatuses:
+  - containerID: containerd://examplecontainerid1234567890abcdef
+    image: debian:latest
+    imageID: docker-pullable://debian@sha256:examplecontainerdigest1234567890abcdef
```
Update this if necessary.

lmktfy commented on content/en/docs/reference/command-line-tools-reference/feature-gates/ImageVolumeWithDigest.md:
```diff
@@ -10,4 +10,5 @@ stages:
     fromVersion: "1.35"
 ---
-Enables impersonation that is constrained to specific requests instead of being all or nothing.
+Report [ImageVolume](/docs/concepts/storage/volumes#image)'s
```
We write `image` in all-lowercase.
```suggestion
For each [`image` volume](/docs/concepts/storage/volumes#image) in a Pod,
report the digest of that image as part of the Pod's status.
```

lmktfy reviewed (commented):
Thanks. It'd be good to document this feature gate, but the PR doesn't feel ready to go in as-is.

iholder101 commented:
/remove-lifecycle rotten

ping @kubernetes/sig-docs-en-reviews @kubernetes/sig-docs-en-owners

natalisucks commented:
thanks for addressing the feedback @iholder101!
/lgtm

reylejano commented:
Thank you, this is useful
/approve
